apps/web/src/server/applications/case/examination-timetable/applications-timetable.controller.js

```diff
@@ -363,18 +363,25 @@ const getTimetableRows = (timetableItem) => {
 
 	const templateType = ExaminationTimetableType?.templateType;
 
+	if (!templateType) {
+		throw new Error(
+			`Template type not found for timetable item type ${ExaminationTimetableType?.name}`
+		);
+	}
+
 	const shouldShowField = (/** @type {string} */ fieldName) =>
-		Object.prototype.hasOwnProperty.call(timetableTemplatesSchema[templateType || 0], fieldName);
+		Object.prototype.hasOwnProperty.call(timetableTemplatesSchema[templateType], fieldName);
 
 	return {
 		id,
 		itemTypeName: ExaminationTimetableType?.name,
 		name,
 		date: shouldShowField('date') ? displayDate(date, { condensed: true }) || '' : null,
-		startDate:
-			shouldShowField('startDate') && startDate
-				? displayDate(startDate, { condensed: true }) || ''
-				: null,
+		startDate: shouldShowField('startDate')
```

Comment:
a short-circuit evaluation (_condition ? option1 : option2_) within another short-circuit is usually very difficult to read: I suggest to rewrite this bit in a "longer" way, maybe above.

also, displayDate() doesn't accept an empty string as argument (you're gonna get the "Invalid time value" error), so this solution _displayDate(startDate || '')_ will not really work when startDate is not defined. 

maybe something like?
`displayDate(startDate, { condensed: true }) || ''`

Reply:
Totally. Will make the relevant changes.